bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java

```diff
@@ -0,0 +1,265 @@
+            return new BigDecimal(Integer.valueOf(type[1], 16));
+        } else if (VALUE_TYPE_FLOAT.equalsIgnoreCase(type[0])) {
+            return parseFloatValue(type[1]);
+        } else {
+            logger.warn("Unknown value type [{}]", type[0]);
+        }
+
+        return BigDecimal.ZERO;
+    }
+
+    /**
+     * Parse the hex coded float value of Senec device and return as BigDecimal
+     *
+     * @param value String with hex float
+     * @return BigDecimal with float value
+     */
+    private static BigDecimal parseFloatValue(String value) {
+        // sample: value = 43E26188
+
+        String binaryString = Long.toString(Long.parseLong(value, 16), 2);
+        while (binaryString.length() < 32) {
+            binaryString = "0" + binaryString;
+        }
+
+        int sign = Character.getNumericValue(binaryString.charAt(0));
+        if (sign == 0) {
+            sign = -1;
+        }
+        sign = sign * -1;
+
+        if (Long.parseLong(binaryString.substring(1), 2) == 0L) {
+            return BigDecimal.ZERO;
+        } else {
+            int exponent = Integer.parseInt(binaryString.substring(1, 9), 2) - 127;
+
+            String mantissa = binaryString.substring(9);
+
+            double significand = 1;
+            for (int i = 1; i < 24; i++) {
+                significand += (1 / Math.pow(2, i)) * Character.getNumericValue(mantissa.charAt(i - 1));
+            }
+
+            return new BigDecimal(significand * Math.pow(2, exponent) * sign);
+        }
+    }
+
+    protected void updatePowerLimitationStatus(Channel channel, boolean status, int duration) {
+        if (this.limitationStatus != null) {
+            if (this.limitationStatus.state == status) {
+                long stateSince = new Date().getTime() - this.limitationStatus.time;
+
+                if (((int) (stateSince / 1000)) < duration) {
+                    // skip updating state (possible flapping state)
+                    return;
+                } else {
+                    logger.debug("{} longer than required duration {}", status, duration);
+                }
+            } else {
+                this.limitationStatus.state = status;
+                this.limitationStatus.time = new Date().getTime();
+
+                // skip updating state (state changed, possible flapping state)
+                return;
+            }
+        } else {
+            this.limitationStatus = new PowerLimitationStatusDTO();
+            this.limitationStatus.state = status;
+        }
+
+        logger.debug("Updating power limitation state {}", status);
+        updateState(channel.getUID(), status ? OnOffType.ON : OnOffType.OFF);
+    }
+
+    protected void updateBatteryState(Channel channel, int code) {
+        updateState(channel.getUID(), new StringType(SenecBatteryStatus.fromCode(code).name()));
+    }
+
+    protected void updateGridPowerValues(Thing thing, BigDecimal gridTotalValue) {
```

Comment:
regarding the updates: see above. And Why do you pass `thing` if you don't use it in the method?